internal/kafka/internal/workers/kafka_mgrs/provisioning_kafkas_mgr.go

```diff
@@ -56,10 +67,77 @@ func (k *ProvisioningKafkaManager) Reconcile() []error {
+				continue
+			}
+		}
 		metrics.UpdateKafkaRequestsStatusSinceCreatedMetric(constants2.KafkaRequestStatusProvisioning, kafka.ID, kafka.ClusterID, time.Since(kafka.CreatedAt))
 	}
 
 	return encounteredErrors
 }
+func (k *ProvisioningKafkaManager) reassignProvisioningKafka(kafka *dbapi.KafkaRequest) error {
+	cluster, e := k.clusterPlacementStrategy.FindCluster(kafka)
+	if e != nil || cluster == nil {
+		msg := fmt.Sprintf("No available cluster found for '%s' Kafka instance in region: '%s'", kafka.InstanceType, kafka.Region)
+		logger.Logger.Infof(msg)
+		return errors.New(fmt.Sprintf("Region %s cannot accept instance type: %s at this moment", kafka.Region, kafka.InstanceType))
+	}
+
+	kafka.ClusterID = cluster.ClusterID
+
+	truncatedKafkaIdentifier := fmt.Sprintf("%s-%s", services.TruncateString(kafka.Name, services.TruncatedNameLen), strings.ToLower(kafka.ID))
+
+	truncatedKafkaIdentifier, replaceErr := services.ReplaceHostSpecialChar(truncatedKafkaIdentifier)
+	if replaceErr != nil {
+		return errors.Wrapf(replaceErr, "generated host is not valid")
+
+	}
+
+	clusterDNS, err := k.clusterService.GetClusterDNS(kafka.ClusterID)
+	if err != nil {
+		return errors.Wrapf(err, "error retrieving cluster DNS for kafka %s", kafka.ID)
+	}
+
+	clusterDNS = strings.Replace(clusterDNS, constants2.DefaultIngressDnsNamePrefix, constants2.ManagedKafkaIngressDnsNamePrefix, 1)
+
+	kafka.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, clusterDNS)
+
+	if k.kafkaConfig.EnableKafkaExternalCertificate {
+		// If we enable KafkaTLS, the bootstrapServerHost should use the external domain name rather than the cluster domain
+		kafka.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, k.kafkaConfig.KafkaDomainName)
+	}
+
+	latestStrimziVersion, latestStrimziVersionErr := cluster.GetLatestAvailableAndReadyStrimziVersion()
+	if latestStrimziVersionErr != nil {
+		return errors.Wrapf(latestStrimziVersionErr, "error finding ready strimzi versions for kafka %s", kafka.ID)
+	}
+	if latestStrimziVersion == nil {
+		return errors.Wrapf(err, "could not find latest strimzi version for kafka %s", kafka.ID)
+	}
+
+	kafka.DesiredStrimziVersion = latestStrimziVersion.Version
+
+	desiredKafkaVersion := latestStrimziVersion.GetLatestKafkaVersion()
+	if desiredKafkaVersion == nil {
+		return errors.Wrapf(err, "failed to get Kafka version for kafka %s", kafka.ID)
+	}
+	kafka.DesiredKafkaVersion = desiredKafkaVersion.Version
+
+	desiredKafkaIBPVersion := latestStrimziVersion.GetLatestKafkaIBPVersion()
+	if desiredKafkaIBPVersion == nil {
+		return errors.Wrapf(err, "failed to get Kafka IBP version for kafka %s", kafka.ID)
+	}
+	kafka.DesiredKafkaIBPVersion = desiredKafkaIBPVersion.Version
+
+	updateErr := k.kafkaService.Update(kafka)
+	if updateErr != nil {
+		return errors.Wrapf(err, "Failed to update kafka %s in provisioning state", kafka.ID)
```

Comment:
See previous comment

Reply:
This should use `updateErr` in place of `err`